src/transforms/kubernetes/kube_config.rs

```diff
@@ -0,0 +1,91 @@
+#![cfg(feature = "kubernetes-integration-tests")]
+
+use dirs;
+use serde::{Deserialize, Serialize};
+use serde_yaml;
+use std::{fs::File, path::PathBuf};
+
+/// Enviorment variable that can containa path to kubernetes config file.
+const CONFIG_PATH: &str = "KUBECONFIG";
+
+/// Loads configuration from local kubeconfig file, the same
+/// one that kubectl uses.
+pub fn load_kube_config() -> Option<Config> {
+    let path = std::env::var(CONFIG_PATH)
+        .ok()
+        .map(PathBuf::from)
+        .or_else(|| dirs::home_dir().map(|home| home.join(".kube").join("config")))?;
+
+    let file = File::open(path)
+        .map_err(|error| debug!(message = "Error opening Kubernetes config file.",%error))
+        .ok()?;
+    serde_yaml::from_reader(file)
+        .map_err(|error| debug!(message = "Error parsing Kubernetes config file.",%error))
+        .ok()
+}
+
+/// Config defines currently relevant data that can be found in
+/// kubernetes config file, the same one that kubectl uses.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Config {
+    pub clusters: Vec<NamedCluster>,
+    pub users: Vec<NamedUserInfo>,
+    pub contexts: Vec<NamedContext>,
+    #[serde(rename = "current-context")]
+    pub current_context: String,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NamedCluster {
+    pub name: String,
+    pub cluster: Cluster,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Cluster {
+    pub server: String,
+    #[serde(rename = "insecure-skip-tls-verify")]
+    pub insecure_skip_tls_verify: Option<bool>,
+    #[serde(rename = "certificate-authority")]
+    pub certificate_authority: Option<String>,
+    #[serde(rename = "certificate-authority-data")]
+    pub certificate_authority_data: Option<String>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NamedUserInfo {
+    pub name: String,
+    pub user: UserInfo,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct UserInfo {
+    pub username: Option<String>,
+    pub password: Option<String>,
+
+    pub token: Option<String>,
+    #[serde(rename = "tokenFile")]
```

Comment:
Seems this parameter has inconsistent naming?

Reply:
Yes, because `token_file` is more inline with Rust naming scheme than `tokenFile`.

Reply:
Agreed! But why is `client-certificate` not `clientCertificate` etc? Is this just a kubernetes thing that I'm overthinking?

Reply:
It's kubernetes thing. For example this
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUN5RENDQWJDZ0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFWTVJNd0VRWURWUVFERXdwcmRXSmwKY201bGRHVnpNQjRYRFRFNU1URXlOREl4TURNMU1sb1hEVEk1TVRFeU1USXhNRE0xTWxvd0ZURVRNQkVHQTFVRQpBeE1LYTNWaVpYSnVaWFJsY3pDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTXZuCjJnNEJQc3I3dTFKQlR0Y3BJanNmTTFOM01VY3d2V3ZJOS85YklYQ2ExVG9waWlYU1hhUUlmRklJVjE0T0dkWUoKeWNLTzBYTEpGOW9oNW1wSmxjc2Znb1NlWDhnRjVVa3QyaVhzdUIwa2tvQ2R0Z3BzQW03TjVSYTA2RlN0ai9HWApPT1J1dXBUZERhcjJXMVVINFpZcTNEekVRaWZGSmVESUNSUVQvcUlIMkFkbzBFTkFOc2NQRWthZ3pPa1hnTGlaCkx3TVhjdldFYTdyN1VySjdJcWRPTVRIRkloSmhSU1pDY1BrMUg4Ky94QTJ1RW9MeWZpbVluQXZxeXNuK0MvMTIKSHpQalVrZFFqMUhDeis4eGNjVXRYOEtmWHhsMVNsd0RKTjRnSTU3eWFHQ3g2UEN5RDV2c3BzemN4UnN4QS9HUgptbS9KUHNXRDNvYjlrRXR6S3BjQ0F3RUFBYU1qTUNFd0RnWURWUjBQQVFIL0JBUURBZ0trTUE4R0ExVWRFd0VCCi93UUZNQU1CQWY4d0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFBd05FQ01ueXllYlg5cU9URFltaWlib0NHSEIKWmozTmVCcldCWG54NThlUmJ4a3RQaFFjbnRPeHJ0VldjclE5aEQrcEIrQmdscWNjYmF5aERzcDJ1aHhJTFNnNAo3U0JZY25VNjlnSEU1cGcyOEh1K21hTk5PRjgxbUc3N3hvT2tDcE41dy92bnRyV0R6czkya3FvYjBmTVdsTENUCjMrTlUwRk5nSS93OEw4V204eGEzMU1oNlpMRVdaSklCV0ZiT0tlbkJNemdFYy83K0R2L1dhaW1PWnQ5a005b0kKQ0FoL21SQVRNZEpJc1Z3QzRUNitkb2gzVCtUSFZ2RmtqYjZBVlJhemVsU2lIWFVOb0dyaE5sM05yOWVrZzZvKwp4Z2ZhY2hNRFF3cnVJU1kyWHNXNjRDbXJ5M3kzZ2RBWTBXSVNnaXZtRVNNNEczQTliWWpma3VWdko3Yz0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
    server: https://647EF4CE61C483DBD4AE629C4E99E771.yl4.eu-west-1.eks.amazonaws.com
  name: kubernetes
- cluster:
    certificate-authority: /home/upwork/.minikube/ca.crt
    server: https://192.168.99.100:8443
  name: minikube
contexts:
- context:
    cluster: kubernetes
    user: aws
  name: aws
- context:
    cluster: minikube
    user: minikube
  name: minikube
current-context: minikube
kind: Config
preferences: {}
users:
- name: aws
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - eks
      - get-token
      - --cluster-name
      - kube14
      - --role
      - arn:aws:iam::816567204970:role/vector_eks
      command: aws2
      env: null
- name: minikube
  user:
    client-certificate: /home/upwork/.minikube/client.crt
    client-key: /home/upwork/.minikube/client.key
```
is my kube config file.